recipes/slepc/meta.yaml

```diff
@@ -0,0 +1,47 @@
+{% set build = 0 %}
+{% set version = '3.7.1' %}
+{% set sha256 = '670216f263e3074b21e0623c01bc0f562fdc0bffcd7bd42dd5d8edbe73a532c2' %}
+{% set mpi = os.environ.get('MPI') or 'mpich' %}
```

Comment:
We can use environment variables in the [build matrix](https://github.com/conda-forge/blas-feedstock/blob/master/conda-forge.yml#L1) on feedstocks, but not on the staged-recipes repo. This is a placeholder in anticipation of that. I added a note about it, and referred to [the mpi metapackage](https://github.com/conda-forge/staged-recipes/pull/1501) before starting out supporting multiple MPIs.